integration/system/options_test.go

```diff
@@ -0,0 +1,118 @@
+package system // import "github.com/docker/docker/integration/system"
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/api"
+	"github.com/docker/docker/api/types"
+	req "github.com/docker/docker/integration-cli/request"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOptionsRoute(t *testing.T) {
+	resp, _, err := req.Do("/", req.Method(http.MethodOptions))
+	require.NoError(t, err)
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+}
+
+func TestGetEnabledCORS(t *testing.T) {
+	resp, body, err := req.Get("/version")
+	require.NoError(t, err)
+	defer body.Close()
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+	// TODO: @runcom incomplete tests, why old integration tests had this headers
+	// and here none of the headers below are in the response?
+	//c.Log(res.Header)
+	//c.Assert(res.Header.Get("Access-Control-Allow-Origin"), check.Equals, "*")
+	//c.Assert(res.Header.Get("Access-Control-Allow-Headers"), check.Equals, "Origin, X-Requested-With, Content-Type, Accept, X-Registry-Auth")
+}
+
+func TestClientVersionOldNotSupported(t *testing.T) {
+	skip.If(t, testEnv.OSType != runtime.GOOS, "Daemon platform doesn't match test platform")
+
+	skip.If(t, api.MinVersion == api.DefaultVersion, "API MinVersion==DefaultVersion")
+
+	v := strings.Split(api.MinVersion, ".")
+	vMinInt, err := strconv.Atoi(v[1])
+	require.NoError(t, err)
+	vMinInt--
+	v[1] = strconv.Itoa(vMinInt)
+	version := strings.Join(v, ".")
+
+	resp, body, err := req.Get("/v" + version + "/version")
+	require.NoError(t, err)
+	defer body.Close()
+	assert.Equal(t, resp.StatusCode, http.StatusBadRequest)
+
+	expected := fmt.Sprintf("client version %s is too old. Minimum supported API version is %s, please upgrade your client to a newer version", version, api.MinVersion)
```

Comment:
I think this should be a unit test.

`integration` tests should be for features/behaviour. We can easily test this error case without running a full daemon.